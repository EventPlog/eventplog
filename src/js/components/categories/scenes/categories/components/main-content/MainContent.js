import React from 'react'
import styled, { css } from 'styled-components'
import Auth from 'js/auth'

//======== Internal Components =========
import Tab from 'js/components/shared/tab'
import CategoriesSection from '../categories-section'
import ContentSection from 'js/components/shared/content-section'
import Sidebar from 'js/components/shared/sidebar'
import { media } from 'js/styles/mixins'

const styles = css`
  .ui.tab {
    box-shadow: none;
    border: 0;
    padding: 1rem 0;
    background: none;
    
    .pagination.secondary.menu {
       display: inline-flex;
    }
  }
  
  .content-header {
    margin: 0 1rem;
    letter-spacing: 0.01rem;
    font-weight: 600;
    font-size: 1.2rem;
    
    ${
      media.phone`
        margin: 0 0 2rem 0;
      `
    } 
  }
  
  .content-panel {
    background: transparent;
    box-shadow: none;
  }
`

const MainContent = ({
  categories = [],
  categories_suggestions = [],
  events_suggestions = [],
  getCategories,
  getCategoriesSuggestions,
  followCategory,
  attendEvent,
  activeIndex,
  className,
}) => {

  const UserCategories = () =>
    <CategoriesSection showCTA={!Auth.isLoggedIn}
                       title="Categories"
      {...{categories, getCategories, followCategory}} />

  const CategoriesSuggestions = () =>
    <CategoriesSection getCategories={getCategoriesSuggestions}
                       title="Categories"
                       followCategory={followCategory}
                       categories={categories_suggestions} />

  const getPanes = () => {
    return [
      {name: `Categories ${Auth.isLoggedIn ? ' I follow' : ' to follow'}`, content: UserCategories },
      {name: `Suggestions`, content: CategoriesSuggestions },
    ]
  }

  return (
    <ContentSection className={`${className} user-categories`}>

      <ContentSection.Body>
        <Tab panes={getPanes()} {...{activeIndex}} />
      </ContentSection.Body>

      <ContentSection.Sidebar>
        <Sidebar.Events events={events_suggestions}
                        attendEvent={attendEvent} />
      </ContentSection.Sidebar>

    </ContentSection>
  )
}

export default styled(MainContent)`${styles}`
