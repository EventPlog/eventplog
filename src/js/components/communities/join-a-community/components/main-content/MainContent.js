import React from 'react'
import styled from 'styled-components'
import defaults from '../../../../../../styles/theme/variables'
import { Form, Label, Select } from 'semantic-ui-react'
import CommunityCard from '../community-card'

const StyledMainContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  
  h2 {
    color: ${defaults.grayLight};
  }
  
  > p {
    margin: 20px 0 50px;
    font-size: 1.2rem;
    font-weight: 300;
  }
  
  .selection {
    width: 100%;
    
    .selection-dropdown {
      display: flex;
      justify-content: space-around;
      margin: 30px 0;
    }
  },
  
  
`
const locationOptions = [
  { key: 'lagos', text: 'Lagos', value: 'lagos' },
  { key: 'abuja', text: 'Lagos', value: 'abuja' },
]

const focusOptions = [
  { key: 'tech', text: 'Technology', value: 'Technology' },
  { key: 'software', text: 'Software', value: 'Software' },
]

const MainContent = ({
  communities = [],
  handleLocationSelect,
  handleFocusChange,
  handleChange
}) => (
  <StyledMainContent className="main-content app-container">
    <h2>Follow your favorite communities</h2>
    <p>To know when they might host an event or share an event-related news</p>
    <div className="selection">
      <div className="selection-dropdown">
        <Form.Field>
          <label>Active location:</label>
          <Select onChange={(e) => handleChange(e, 'gender')}
                  defaultValue='Lagos'
                  placeholder='State' options={locationOptions} />
        </Form.Field>

        <Form.Field>
          <label>Community focus: </label>
          <Select onChange={(e) => handleChange(e, 'gender')}
                  defaultValue='Technology'
                  placeholder='Gender' options={focusOptions} />
        </Form.Field>
      </div>
      <div className="options">
        { !!communities && communities.map(community =>  (
          <CommunityCard {...community} />
        ))
        }
      </div>
    </div>
  </StyledMainContent>
)

export default MainContent
