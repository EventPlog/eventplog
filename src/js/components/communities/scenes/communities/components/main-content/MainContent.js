import React from 'react'
import styled from 'styled-components'
import { media } from 'js/styles/mixins'
import CommunitiesSection from '../communities-section'
import EventsSection from '../events-section'
import ContentSection from 'js/components/shared/content-section'


const MainContent = ({
  communities = [],
  communities_suggestions = [],
  events_suggestions = []
}) => {
  return (
    <ContentSection className="user-communities">

      <ContentSection.Body>
        <CommunitiesSection title="My communities"
                            showCTA={false}
          {...{communities}} />
        <CommunitiesSection title="Related communities"
                            communities={communities_suggestions} />
      </ContentSection.Body>

      <ContentSection.Sidebar>
        <EventsSection events={events_suggestions} />
      </ContentSection.Sidebar>

    </ContentSection>
  )
}

export default MainContent