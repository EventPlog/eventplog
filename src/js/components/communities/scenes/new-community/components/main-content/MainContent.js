import React from 'react'
import styled from 'styled-components'
import defaults from 'js/styles/theme/variables'
import ContentBeforeCommunitySubmit from '../content-before-community-submit'
import ContentAfterCommunitySubmit from '../content-after-community-submit'

const StyledMainContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  align-items: center;
  
  h3 {
    color: ${defaults.grayMedium};
  }
  
  > p {
    margin: 20px 0 50px;
    font-size: 1.2rem;
    font-weight: 300;
    line-height: 1.9rem;
  }
  
`


const MainContent = ({
  community,
  error,
  loading,
  communityCreated,
  handleChange,
  submitCommunity
}) => (
    <StyledMainContent className="main-content app-container">
      { !communityCreated &&
        <ContentBeforeCommunitySubmit {...{loading, error, community, handleChange, submitCommunity}} />}

      { communityCreated && <ContentAfterCommunitySubmit {...{community, handleChange}} /> }
    </StyledMainContent>
    )

    export default MainContent
