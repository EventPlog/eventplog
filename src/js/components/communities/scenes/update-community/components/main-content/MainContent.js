import React from 'react';
import styled from 'styled-components';
import { Form, label, Message, Checkbox, Icon } from 'semantic-ui-react'

import ContentPanel from 'js/components/shared/content-panel'
import Input from 'js/components/shared/input'
import Button from 'js/components/shared/button'
import Select from 'js/components/shared/select'
import color from 'js/styles/theme/variables'

const StyleEventUpdate = styled.div`
  
  > img {
    width: 100%;
  }
  
  form.ui.form {
  
    .fields {
      margin-bottom: 2rem;
    }
    
    .checkbox {
      margin-bottom: 2rem;
    }
    
    .date-time-picker {
    }
  
  }
  
  .cancel {
    margin-left: 1rem;
  }
  
  .same-line {
    display: flex;
    align-items: center;
  }
  
  input[name='slug'] {
    text-align: right;
  }
  
`

const colorOptions = [
  { key: 'default', value: color.primary, icon: 'point purple', text: 'default' },
  { key: 'blue', value: color.blue, icon: 'point blue', text: 'blue' },
  { key: 'red', value: color.red, icon: 'point red', text: 'red' },
  { key: 'green', value: color.green, icon: 'point green', text: 'green' },
  { key: 'pink', value: color.pink, icon: 'point pink', text: 'pink' },
  { key: 'orange', value: color.orange, icon: 'point orange', text: 'orange' },
  { key: 'yellow', value: color.yellow, icon: 'point yellow', text: 'yellow' },
]

const EventUpdate = ({
  community = {},
  loading,
  success,
  handleChange,
  handleSubmit,
  slug_check = {},
  checkForValidSlug,
}) => {
  const {
    id,
    name,
    description,
    featured_image,
    logo,
    link,
    slug,
    topic_interests = [],
    brand_color,
  } = community
  return (
    <StyleEventUpdate className="main-content">
      <ContentPanel title="Edit Community">
        <Form loading={loading} success={success}>
          <Message
            success
            header='Yes!'
            content="You've successfully updated this community"
          />

          <Form.Field>
            <label>Name</label>
            <Input name="name"
                   value={name}
                   placeholder='Awesome event'
                   onChange={handleChange}/>
          </Form.Field>

          <Form.Field>
            <label>One-line Description</label>
            <Input name="description"
                   value={description}
                   maxLength={70}
                   placeholder='An community of awesome people' onChange={handleChange}/>
          </Form.Field>

          <Form.Field>
            <label>Slug</label>
            {slug_check.valid &&
              <div className="success green">Slug is available!</div>}
            {slug_check.error &&
              <div className="error red">{slug_check.error}</div>}
            {slug_check.loading &&
              <div className="">Checking for availability  <Icon loading name='asterisk' /></div>}

            <Form.Field widths="equal" className="same-line">
              eventplog.com/c/
              <Input name="slug"
                     value={slug}
                     onBlur={checkForValidSlug}
                     disabled={slug_check.loading}
                     placeholder='something' onChange={handleChange}/>
            </Form.Field>
          </Form.Field>

          <Form.Field>
            <label>Link to Group</label>
            <Input name="link"
                   value={link}
                   placeholder='http://someplace.com' onChange={handleChange}/>
          </Form.Field>

          <Form.Field>
            <label>Link to logo</label>
            <Input name="logo"
                   value={logo}
                   placeholder='http://someplace.com' onChange={handleChange}/>
          </Form.Field>

          <Form.Field>
            <label>Featured image url</label>
            <Input name="featured_image"
                   value={featured_image}
                   placeholder='http://someplace.com' onChange={handleChange}/>
          </Form.Field>

          <Form.Field>
            <label>Interests (separated by commas)</label>
            <Input name="topic_interests"
                   value={topic_interests.join(',')}
                   placeholder='technology, food'
                   onChange={(e) => handleChange(e, {
                     name: e.target.name,
                     value: e.target.value.split(',')
                   })}/>
          </Form.Field>

          <Form.Field>
            <label>Brand Color</label>
            <Select options={colorOptions}
                    name="brand_color"
                    value={brand_color}
                    defaultValue={color.activeLink}
                    onChange={handleChange}/>

          </Form.Field>

          {/*<Form.Group widths="equal">*/}
          {/*</Form.Group>*/}

          <Form.Field>
            <Checkbox checked label='Make this community public' />
          </Form.Field>

          <Form.Group>
            <Button inverted type='submit' onClick={handleSubmit} disabled={slug_check.error}>
              Save
            </Button>

            <Button.Link className="cancel" to={`/communities/${community.id}`}>
              Back to Community Page
            </Button.Link>
          </Form.Group>
        </Form>
      </ContentPanel>
    </StyleEventUpdate>
  )
}

export default EventUpdate;
