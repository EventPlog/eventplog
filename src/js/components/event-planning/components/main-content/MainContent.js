import React from 'react'
import ReactMarkdown from 'react-markdown'
import styled from 'styled-components'

// internal components
import Tab from 'js/components/shared/tab';
import TodoItems from 'js/components/todo-items';
import ContentPanel from 'js/components/shared/content-panel'
import ContentEditable from 'js/components/shared/content-editable'
import Loading from 'js/components/shared/loading'

const StyledEventPlanning = styled.div`
  .tasks .content-body {
    margin: 2rem 0;
  }
  
  ul {
    list-style-type: initial;
    padding-left: 1rem;
  }
`

const EventPlanningMainContent = ({todo_items = {data: []}, ...props}) => {
  const getPanes = () => {
    const todoData = todo_items.data.filter(item => item.recipient_id == props.user.id)
    const userTodo = {...todo_items, data: todoData}
    return [
      {name: `My Task (${todoData.length})`, content: (props) => <TodoItems {...{...props, todo_items: userTodo}} /> },
      {name: `All Tasks (${todo_items.data.length})`, content: (props) => <TodoItems {...{...props, todo_items}} /> },
    ]
  }

  const {
    event_checklist = {},
    handleChange,
    handleSubmit,
    pending = [],
    in_progress = [],
    completed = [],
    loading,
    error,
  } = props
  const {description} = event_checklist

  if (loading) return <Loading />
  if (error) return <Loading.Error msg={error} />

  return (
    <StyledEventPlanning>
      <ContentPanel title="Goal(s)">
        <ContentEditable propName="description"
                         type="textarea"
                         canEdit={true}
                         defaultValue={description}
                         onChange={handleChange}
                         onSubmit={handleSubmit}>
          <ReactMarkdown source={description || 'Click to edit. In markdown, if you wish :)'} />
        </ContentEditable>
      </ContentPanel>
      <ContentPanel className="tasks" title="Tasks">
        <Tab panes={getPanes()} {...props} />
      </ContentPanel>
    </StyledEventPlanning>

  )

}

export default EventPlanningMainContent;
