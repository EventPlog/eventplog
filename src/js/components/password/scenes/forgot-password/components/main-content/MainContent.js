import React from 'react'
import styled from 'styled-components'
import defaults from '../../../../../../../theme/variables'
import ContentBeforeEmailSubmit from '../content-before-email-submit'
import ContentAfterEmailSubmit from '../content-after-email-submit'

const StyledMainContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  
  h3 {
    color: ${defaults.grayMedium};
  }
  
  > p {
    margin: 20px 0 50px;
    font-size: 1.2rem;
    font-weight: 300;
  }
  
  .selection {
    width: 100%;
    
    .selection-dropdown {
      display: flex;
      justify-content: space-around;
      margin: 30px 0;
    }
  },
  
`


const MainContent = ({
  email,
  emailSubmitted,
  handleChange,
  submitEmail
}) => (
    <StyledMainContent className="main-content app-container">
      <h3>Forgot your password?</h3>
      { !emailSubmitted && <ContentBeforeEmailSubmit {...{email, handleChange}} />}
      { emailSubmitted && <ContentAfterEmailSubmit {...{email, handleChange}} /> }
    </StyledMainContent>
    )

    export default MainContent
