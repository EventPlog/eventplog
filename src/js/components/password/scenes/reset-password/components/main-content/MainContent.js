import React from 'react'
import styled from 'styled-components'
import defaults from '../../../../../../styles/theme/variables'
import ContentBeforePasswordChange from '../content-before-password-change'
import ContentAfterPasswordChange from '../content-after-password-change'

const StyledMainContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  align-items: center;
  
  h3 {
    color: ${defaults.grayMedium};
  }
  
  > p {
    margin: 20px 0 50px;
    font-size: 1.2rem;
    font-weight: 300;
  }
  
`
const headerText = (passwordChanged) => (
  passwordChanged
    ? 'Success!!!'
    : 'One step left...'
)

const MainContent = ({
  token,
  password,
  error,
  loading,
  passwordChanged,
  handleChange,
  resetPassword
}) => (
    <StyledMainContent className="main-content app-container">
      <h3>{headerText(passwordChanged)}</h3>
      { !passwordChanged &&
        <ContentBeforePasswordChange {...{loading, error, password, handleChange, resetPassword}} />}

      { passwordChanged && <ContentAfterPasswordChange {...{password, handleChange}} /> }
    </StyledMainContent>
    )

    export default MainContent
