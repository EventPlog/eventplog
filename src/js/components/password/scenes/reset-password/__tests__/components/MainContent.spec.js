import React from 'react';
import MainContent from '../../components/main-content'
import { shallow } from 'enzyme';

describe('main content - join a community component', () => {
  it('should render correctly', () => {
    const wrapper = shallow( <MainContent /> );
    expect(wrapper).toMatchSnapshot()
  })

  it('should show the right message', () => {
    const wrapper = shallow(<MainContent />)
    expect(true).toBe(true)
  })
});