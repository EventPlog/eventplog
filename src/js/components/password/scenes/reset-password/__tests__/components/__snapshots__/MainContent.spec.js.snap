// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main content - join a community component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h2>
          Follow your favorite communities
</h2>,
        <p>
          To know when they might host an event or share an event-related news
</p>,
        <div
          className="selection"
>
          <div
                    className="selection-dropdown"
          >
                    <FormField>
                              <label>
                                        Active location:
                              </label>
                              <Select
                                        defaultValue="Lagos"
                                        onChange={[Function]}
                                        options={
                                                  Array [
                                                            Object {
                                                              "key": "lagos",
                                                              "text": "Lagos",
                                                              "value": "lagos",
                                                            },
                                                            Object {
                                                              "key": "abuja",
                                                              "text": "Lagos",
                                                              "value": "abuja",
                                                            },
                                                          ]
                                        }
                                        placeholder="State"
                              />
                    </FormField>
                    <FormField>
                              <label>
                                        Community focus: 
                              </label>
                              <Select
                                        defaultValue="Technology"
                                        onChange={[Function]}
                                        options={
                                                  Array [
                                                            Object {
                                                              "key": "tech",
                                                              "text": "Technology",
                                                              "value": "Technology",
                                                            },
                                                            Object {
                                                              "key": "software",
                                                              "text": "Software",
                                                              "value": "Software",
                                                            },
                                                          ]
                                        }
                                        placeholder="Gender"
                              />
                    </FormField>
          </div>
</div>,
      ],
      "className": "main-content app-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Follow your favorite communities",
        },
        "ref": null,
        "rendered": "Follow your favorite communities",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To know when they might host an event or share an event-related news",
        },
        "ref": null,
        "rendered": "To know when they might host an event or share an event-related news",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="selection-dropdown"
>
            <FormField>
                        <label>
                                    Active location:
                        </label>
                        <Select
                                    defaultValue="Lagos"
                                    onChange={[Function]}
                                    options={
                                                Array [
                                                            Object {
                                                              "key": "lagos",
                                                              "text": "Lagos",
                                                              "value": "lagos",
                                                            },
                                                            Object {
                                                              "key": "abuja",
                                                              "text": "Lagos",
                                                              "value": "abuja",
                                                            },
                                                          ]
                                    }
                                    placeholder="State"
                        />
            </FormField>
            <FormField>
                        <label>
                                    Community focus: 
                        </label>
                        <Select
                                    defaultValue="Technology"
                                    onChange={[Function]}
                                    options={
                                                Array [
                                                            Object {
                                                              "key": "tech",
                                                              "text": "Technology",
                                                              "value": "Technology",
                                                            },
                                                            Object {
                                                              "key": "software",
                                                              "text": "Software",
                                                              "value": "Software",
                                                            },
                                                          ]
                                    }
                                    placeholder="Gender"
                        />
            </FormField>
</div>,
          "className": "selection",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FormField>
                <label>
                                Active location:
                </label>
                <Select
                                defaultValue="Lagos"
                                onChange={[Function]}
                                options={
                                                Array [
                                                                Object {
                                                                  "key": "lagos",
                                                                  "text": "Lagos",
                                                                  "value": "lagos",
                                                                },
                                                                Object {
                                                                  "key": "abuja",
                                                                  "text": "Lagos",
                                                                  "value": "abuja",
                                                                },
                                                              ]
                                }
                                placeholder="State"
                />
</FormField>,
              <FormField>
                <label>
                                Community focus: 
                </label>
                <Select
                                defaultValue="Technology"
                                onChange={[Function]}
                                options={
                                                Array [
                                                                Object {
                                                                  "key": "tech",
                                                                  "text": "Technology",
                                                                  "value": "Technology",
                                                                },
                                                                Object {
                                                                  "key": "software",
                                                                  "text": "Software",
                                                                  "value": "Software",
                                                                },
                                                              ]
                                }
                                placeholder="Gender"
                />
</FormField>,
            ],
            "className": "selection-dropdown",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label>
                    Active location:
</label>,
                  <Select
                    defaultValue="Lagos"
                    onChange={[Function]}
                    options={
                                        Array [
                                                            Object {
                                                              "key": "lagos",
                                                              "text": "Lagos",
                                                              "value": "lagos",
                                                            },
                                                            Object {
                                                              "key": "abuja",
                                                              "text": "Lagos",
                                                              "value": "abuja",
                                                            },
                                                          ]
                    }
                    placeholder="State"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Active location:",
                  },
                  "ref": null,
                  "rendered": "Active location:",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "defaultValue": "Lagos",
                    "onChange": [Function],
                    "options": Array [
                      Object {
                        "key": "lagos",
                        "text": "Lagos",
                        "value": "lagos",
                      },
                      Object {
                        "key": "abuja",
                        "text": "Lagos",
                        "value": "abuja",
                      },
                    ],
                    "placeholder": "State",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label>
                    Community focus: 
</label>,
                  <Select
                    defaultValue="Technology"
                    onChange={[Function]}
                    options={
                                        Array [
                                                            Object {
                                                              "key": "tech",
                                                              "text": "Technology",
                                                              "value": "Technology",
                                                            },
                                                            Object {
                                                              "key": "software",
                                                              "text": "Software",
                                                              "value": "Software",
                                                            },
                                                          ]
                    }
                    placeholder="Gender"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Community focus: ",
                  },
                  "ref": null,
                  "rendered": "Community focus: ",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "defaultValue": "Technology",
                    "onChange": [Function],
                    "options": Array [
                      Object {
                        "key": "tech",
                        "text": "Technology",
                        "value": "Technology",
                      },
                      Object {
                        "key": "software",
                        "text": "Software",
                        "value": "Software",
                      },
                    ],
                    "placeholder": "Gender",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h2>
            Follow your favorite communities
</h2>,
          <p>
            To know when they might host an event or share an event-related news
</p>,
          <div
            className="selection"
>
            <div
                        className="selection-dropdown"
            >
                        <FormField>
                                    <label>
                                                Active location:
                                    </label>
                                    <Select
                                                defaultValue="Lagos"
                                                onChange={[Function]}
                                                options={
                                                            Array [
                                                                        Object {
                                                                          "key": "lagos",
                                                                          "text": "Lagos",
                                                                          "value": "lagos",
                                                                        },
                                                                        Object {
                                                                          "key": "abuja",
                                                                          "text": "Lagos",
                                                                          "value": "abuja",
                                                                        },
                                                                      ]
                                                }
                                                placeholder="State"
                                    />
                        </FormField>
                        <FormField>
                                    <label>
                                                Community focus: 
                                    </label>
                                    <Select
                                                defaultValue="Technology"
                                                onChange={[Function]}
                                                options={
                                                            Array [
                                                                        Object {
                                                                          "key": "tech",
                                                                          "text": "Technology",
                                                                          "value": "Technology",
                                                                        },
                                                                        Object {
                                                                          "key": "software",
                                                                          "text": "Software",
                                                                          "value": "Software",
                                                                        },
                                                                      ]
                                                }
                                                placeholder="Gender"
                                    />
                        </FormField>
            </div>
</div>,
        ],
        "className": "main-content app-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Follow your favorite communities",
          },
          "ref": null,
          "rendered": "Follow your favorite communities",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To know when they might host an event or share an event-related news",
          },
          "ref": null,
          "rendered": "To know when they might host an event or share an event-related news",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="selection-dropdown"
>
              <FormField>
                            <label>
                                          Active location:
                            </label>
                            <Select
                                          defaultValue="Lagos"
                                          onChange={[Function]}
                                          options={
                                                        Array [
                                                                      Object {
                                                                        "key": "lagos",
                                                                        "text": "Lagos",
                                                                        "value": "lagos",
                                                                      },
                                                                      Object {
                                                                        "key": "abuja",
                                                                        "text": "Lagos",
                                                                        "value": "abuja",
                                                                      },
                                                                    ]
                                          }
                                          placeholder="State"
                            />
              </FormField>
              <FormField>
                            <label>
                                          Community focus: 
                            </label>
                            <Select
                                          defaultValue="Technology"
                                          onChange={[Function]}
                                          options={
                                                        Array [
                                                                      Object {
                                                                        "key": "tech",
                                                                        "text": "Technology",
                                                                        "value": "Technology",
                                                                      },
                                                                      Object {
                                                                        "key": "software",
                                                                        "text": "Software",
                                                                        "value": "Software",
                                                                      },
                                                                    ]
                                          }
                                          placeholder="Gender"
                            />
              </FormField>
</div>,
            "className": "selection",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FormField>
                  <label>
                                    Active location:
                  </label>
                  <Select
                                    defaultValue="Lagos"
                                    onChange={[Function]}
                                    options={
                                                      Array [
                                                                        Object {
                                                                          "key": "lagos",
                                                                          "text": "Lagos",
                                                                          "value": "lagos",
                                                                        },
                                                                        Object {
                                                                          "key": "abuja",
                                                                          "text": "Lagos",
                                                                          "value": "abuja",
                                                                        },
                                                                      ]
                                    }
                                    placeholder="State"
                  />
</FormField>,
                <FormField>
                  <label>
                                    Community focus: 
                  </label>
                  <Select
                                    defaultValue="Technology"
                                    onChange={[Function]}
                                    options={
                                                      Array [
                                                                        Object {
                                                                          "key": "tech",
                                                                          "text": "Technology",
                                                                          "value": "Technology",
                                                                        },
                                                                        Object {
                                                                          "key": "software",
                                                                          "text": "Software",
                                                                          "value": "Software",
                                                                        },
                                                                      ]
                                    }
                                    placeholder="Gender"
                  />
</FormField>,
              ],
              "className": "selection-dropdown",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label>
                      Active location:
</label>,
                    <Select
                      defaultValue="Lagos"
                      onChange={[Function]}
                      options={
                                            Array [
                                                                  Object {
                                                                    "key": "lagos",
                                                                    "text": "Lagos",
                                                                    "value": "lagos",
                                                                  },
                                                                  Object {
                                                                    "key": "abuja",
                                                                    "text": "Lagos",
                                                                    "value": "abuja",
                                                                  },
                                                                ]
                      }
                      placeholder="State"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Active location:",
                    },
                    "ref": null,
                    "rendered": "Active location:",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "defaultValue": "Lagos",
                      "onChange": [Function],
                      "options": Array [
                        Object {
                          "key": "lagos",
                          "text": "Lagos",
                          "value": "lagos",
                        },
                        Object {
                          "key": "abuja",
                          "text": "Lagos",
                          "value": "abuja",
                        },
                      ],
                      "placeholder": "State",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label>
                      Community focus: 
</label>,
                    <Select
                      defaultValue="Technology"
                      onChange={[Function]}
                      options={
                                            Array [
                                                                  Object {
                                                                    "key": "tech",
                                                                    "text": "Technology",
                                                                    "value": "Technology",
                                                                  },
                                                                  Object {
                                                                    "key": "software",
                                                                    "text": "Software",
                                                                    "value": "Software",
                                                                  },
                                                                ]
                      }
                      placeholder="Gender"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Community focus: ",
                    },
                    "ref": null,
                    "rendered": "Community focus: ",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "defaultValue": "Technology",
                      "onChange": [Function],
                      "options": Array [
                        Object {
                          "key": "tech",
                          "text": "Technology",
                          "value": "Technology",
                        },
                        Object {
                          "key": "software",
                          "text": "Software",
                          "value": "Software",
                        },
                      ],
                      "placeholder": "Gender",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
